FEAT : enter-room / leave-room 이벤트에 webRTC 연결/해제 기능 개발#62

jn33-dev opened the pull request:
---
name: Pull Request
about: 게임 방에 출입할 때, P2P webRTC 연결/해제 처리
title: FEAT : enter-room / leave-room 이벤트에 webRTC 연결/해제 기능 개발#62
labels: 'backend', 'enhancement'
assignees: 최예나

---

**작업한 내용**

- webrtc-ice
- webrtc-offer
- webrtc-answer
- webrtc-leave
- update-userstream
이벤트 추가


**REVIEW POINT**
- webrtc-ice
- webrtc-offer
- webrtc-answer
_**네 가지 이벤트의 경우, 이벤트를 받을 socketId보다 해당 방에 이벤트를 emit하면 되는거 아닌지 프런트와 확인이 필요**_

close issue #61

@pds0309

jn33-dev committed:
FIX : player.service 레포지토리 주입 및 로직 수정 #58

Changes to be committed:
	renamed:    project/src/games/dto/enter-room.dto.ts -> project/src/games/dto/enter-room.request.dto.ts
	renamed:    project/src/users/dto/login-user.dto.ts -> project/src/games/dto/socketId-map.request.dto.ts
	modified:   project/src/games/entities/player.entity.ts
	modified:   project/src/games/entities/room.entity.ts
	modified:   project/src/games/entities/socketIdMap.entity.ts
	modified:   project/src/games/games.gateway.ts
	modified:   project/src/games/players.service.ts
	modified:   project/src/games/room.service.ts
	new file:   project/src/users/dto/authorization.dto.ts
	modified:   project/src/users/users.controller.ts
	modified:   project/src/users/users.service.ts

jn33-dev committed:
DATA : 데이터 엔티티 수정, ERD 수정 #58

- socketIdMap에서 currentRoom 컬럼을 빼고, player 테이블에서 roomId를 조회하는 방식으로 변경 (socketIdMap 조회할때, 자동 조회)
- room 데이터 조회할때, player 데이터를 자동으로 같이 조회하도록 설정 변경
- tokenMap 테이블에서, userId를 PK로 변경
- cacade 옵션 : player에서 socketIdMap 삭제될때 & room 삭제될때, 자동 삭제
Changes to be committed:
	modified:   dataSource.ts
	modified:   src/app.module.ts
	modified:   src/games/dto/create-room.request.dto.ts
	modified:   src/games/dto/room.data.dto.ts
	modified:   src/games/dto/socketId-map.request.dto.ts
	modified:   src/games/entities/player.entity.ts
	modified:   src/games/entities/room.entity.ts
	modified:   src/games/entities/socketIdMap.entity.ts
	modified:   src/games/games.gateway.ts
	modified:   src/games/players.service.ts
	modified:   src/games/room.service.ts
	modified:   src/users/entities/token-map.entity.ts

jn33-dev committed:
REFACTOR : create-room event 레포지토리 주입하여 수정 #58

Changes to be committed:
	renamed:    src/games/dto/room.data.dto.ts -> src/games/dto/room.data.insert.dto.ts
	modified:   src/games/dto/roomInfoToMain.dto.ts
	modified:   src/games/dto/roomInfoToRoom.dto.ts
	modified:   src/games/dto/socketId-map.request.dto.ts
	modified:   src/games/entities/room.entity.ts
	modified:   src/games/entities/socketIdMap.entity.ts
	modified:   src/games/games.gateway.ts
	modified:   src/games/players.service.ts
	modified:   src/games/room.service.ts
	modified:   src/users/entities/token-map.entity.ts
	modified:   src/users/users.controller.ts
	modified:   src/users/users.service.ts

jn33-dev committed:
REFACTOR : enter-room, leave-room, log-out event 레포지토리 주입 후 수정 #58

Changes to be committed:
	modified:   src/games/dto/enter-room.request.dto.ts
	new file:   src/games/dto/evnet-user.info.dto.ts
	modified:   src/games/dto/roomInfoToRoom.dto.ts
	modified:   src/games/dto/socketId-map.request.dto.ts
	modified:   src/games/entities/player.entity.ts
	modified:   src/games/entities/room.entity.ts
	modified:   src/games/entities/socketIdMap.entity.ts
	modified:   src/games/games.gateway.ts
	modified:   src/games/players.service.ts
	modified:   src/games/room.service.ts
	new file:   src/games/util/event.user.info.constructor.ts
	new file:   src/games/util/participants-list.mapper.ts
	new file:   src/games/util/update-room.announcer.ts
	modified:   src/users/entities/token-map.entity.ts

jn33-dev committed:
FIX : socketIdMap에서 유저 정보 조회할때, playerInfo>roomInfo 같이 조회하도록 수정 #58

Changes to be committed:
	modified:   src/games/games.gateway.ts
	modified:   src/games/players.service.ts

jn33-dev committed:
FEAT : leave-room / log-out / disconnect event에 방 삭제 로직 추가 #61

Changes to be committed:
	modified:   src/common/exceptionFilters/ws-exception.filter.ts
	modified:   src/games/games.gateway.ts
	modified:   src/games/players.service.ts
	modified:   src/games/room.service.ts
	deleted:    src/games/util/update-room.announcer.ts
	modified:   src/main.ts

jn33-dev committed:
FEAT : leave-room / log-out / disconnect event에 방 삭제 로직 추가 #61

- 유저가 방 나갈때, 마지막 사람이면 방 폭파
- 다른 유저가 남아 있고, 유저가 방장이었다면, 방장 변경
- entities 중 관계맺은 column의 이름과 타입 수정
Changes to be committed:
	modified:   src/games/dto/socketId-map.request.dto.ts
	modified:   src/games/entities/player.entity.ts
	modified:   src/games/entities/room.entity.ts
	modified:   src/games/entities/socketIdMap.entity.ts
	modified:   src/games/games.gateway.ts
	modified:   src/games/players.service.ts
	modified:   src/games/room.service.ts
	modified:   src/games/util/event.user.info.constructor.ts
	modified:   src/users/entities/token-map.entity.ts
	modified:   src/users/entities/user.entity.ts
	modified:   src/users/users.controller.ts
	modified:   src/users/users.service.ts

RESOLVE : issue #58, #61

jn33-dev committed:
FEAT : enter-room / leave-room 이벤트에 webRTC 연결/해제 기능 개발#62

Changes to be committed:
	modified:   dist/src/games/games.gateway.d.ts
	modified:   dist/src/games/games.gateway.js
	modified:   dist/src/games/games.gateway.js.map
	modified:   dist/tsconfig.build.tsbuildinfo
	modified:   src/games/games.gateway.ts

pds0309 commented:
그리고 결국 p2p연결이기때문에 들어온사람에게 내 미디어정보를 전달하는걸로 시작해서 쭉 상호작용하기때문에 일부러 저런식으로(소켓 아이디로) 구성했던것 같습니다!

jn33-dev committed:
FIX : webRTC-ice, offer, answer 이벤트 emit 대상 수정 #62

- 프런트 요청사항 반영하여 수정
Changes to be committed:
	modified:   src/app.module.ts
	modified:   src/games/games.gateway.ts
	modified:   src/users/entities/token-map.entity.ts

RESOLVE : issue #62
